packages/c-flex/src/flex.ts

```diff
@@ -0,0 +1,59 @@
+import { h, defineComponent, PropType, computed } from 'vue'
+import { chakra, DOMElements, ThemingProps } from '@chakra-ui/vue-system'
+
+export type FlexProps = 'align' | 'justify' | 'wrap' | 'direction' | 'size'
```

Comment:
Here, it's best to use an [interface](https://www.typescriptlang.org/docs/handbook/interfaces.html) for the prop types of the component.

In this case you used a [Union](https://www.typescriptlang.org/docs/handbook/unions-and-intersections.html) of strings type. (That means a prop of type `FlexProps` can only accept values of `align` | `justify` ... & `size`. And this is not the desired result.

Here's how to fix this: 👍🏽

1. Import the `SystemProps` interface from `@chakra-ui/vue-system`. It contains the types for all the possible Chakra UI Vue style props you can use. 
```ts
import { SystemProps } from '@chakra-ui/vue-system'
```

2. Declare an interface for the `FlexProps`:

```ts
export interface FlexProps {
  /** We'll declare our prop types here
}
```

3. Bind set the types for the properties inside the interface.
```ts
export interface FlexProps {
  /**
   * Shorthand for `alignItems` style prop
   * @type SystemProps["alignItems"]
   */
  align?: SystemProps["alignItems"]

  /**
   * Shorthand for `justifyContent` style prop
   * @type SystemProps["justifyContent"]
   */
  justify?: SystemProps["justifyContent"]
  
  // You can add the rest of the prop types here fro wrap, direction, etc. The `SystemProps` interface is quite self documenting.
}

```

4. When declaring the `CFlex` component props, you can rely on your new shiny `FlexProps` interface.

```ts
const CFlex = defineComponent({
  props: {
    // ...
    align: [Object, String, Array] as PropType<FlexProps['align']>,
    justify: [Object, String, Array] as PropType<FlexProps['justify']>,
    // ...
  },
  // ..
}
```